plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/PipelineArgumentsTab.java

```diff
@@ -262,7 +269,9 @@ public void clear() {
 
   @Override
   public void setDefaults(ILaunchConfigurationWorkingCopy configuration) {
-    launchConfiguration = PipelineLaunchConfiguration.createDefault();
+    project = findProject(configuration);
+    MajorVersion version = getMajorVersion(project);
```

Comment:
Set it to the field `majorVersion`?

Reply:
Not appropriate here — `setDefaults()` is called when a new launch configuration is created, e.g., using the create-new button from the Run/Debug Configurations dialog.  It's intended to pre-populate the launch config with suitable defaults.

Reply:
If so, is it OK that we save `project` just above (and `launchConfiguration` below)? It looks inconsistent that you save `project` and `launchConfiguration` while you don't for `majorVersion`.